app/models/user.rb

```diff
@@ -9,9 +9,67 @@ class User < ApplicationRecord
   validates :first_name, :last_name, :username, presence: true
   validates :username, uniqueness: true
 
-  def data
-    User.all.as_json.find do |user|
-      user['username'] == username
+  has_many :active_friendships, foreign_key: :active_friend_id, dependent: :destroy, class_name: 'Friendship'
+  has_many :passive_friendships, foreign_key: :passive_friend_id, dependent: :destroy, class_name: 'Friendship'
+  has_many :active_friends, through: :passive_friendships, source: :active_friend, dependent: :destroy
+  has_many :passive_friends, through: :active_friendships, source: :passive_friend, dependent: :destroy
+
+  default_scope { order(created_at: :asc) }
+
+  def friends
+    User.where(id: active_friendships.where(confirmed: true).pluck(:passive_friend_id))
```

Comment:
I really like this solution <3

Reply:
Thanks! Haha i remember you challenged our "old" way of setting the friends (which is to have an array where we add `active_friends` + `passive_friends`). I realized a db collection is much faster indeed! Thanks for that Emilia! :)